packages/botonic-cli/src/commands/new.ts

```diff
@@ -146,4 +95,73 @@ Creating...
       },
     ])
   }
+
+  async resolveSelectedProject(
+    projectName: string | undefined
+  ): Promise<BotonicProject | undefined> {
+    if (!projectName)
+      return await this.selectBotName().then(userInput =>
+        this.examples.find(p => p.description === userInput.botName)
+      )
+    return this.examples.find(p => p.name === projectName)
+  }
+
+  async downloadSelectedProjectIntoPath(
+    selectedProject: BotonicProject,
+    path: string
+  ): Promise<void> {
+    if (existsSync(path)) rimraf.sync(path)
+    const spinner = ora({
+      text: 'Downloading files...',
+      spinner: 'bouncingBar',
+    }).start()
+    try {
+      await fetchRepoDir({
+        src: selectedProject.uri,
+        dir: path,
+      })
+      spinner.succeed()
+    } catch (e) {
+      spinner.fail()
+      const error = `Downloading Project: ${selectedProject.name}: ${String(e)}`
+      trackError(error)
+      throw new Error(error)
+    }
+  }
+
+  async installProjectDependencies(path: string): Promise<void> {
+    process.chdir(path)
+    const spinner = ora({
+      text: 'Installing dependencies...',
+      spinner: 'bouncingBar',
+    }).start()
+    try {
+      await exec('npm install')
+      spinner.succeed()
+    } catch (e) {
+      spinner.fail()
+      const error = `Installing dependencies: ${String(e)}`
+      trackError(error)
+      throw new Error(error)
+    }
+  }
+
+  showFeedback(selectedProject: BotonicProject, path: string): void {
+    const chdirCmd = bold(`cd ${path}`)
```

Comment:
I added https://wiki.c2.com/?DeclareVariablesAtFirstUse to our https://docs.google.com/document/d/1-qd2FdXbAe7XEbdTGyACeCs9HRCauCogocfEU-h-n-I/edit
I think you can get rid of the constants. they just make the method longer and difficult to read. For instance, sucessText is declared last but used first.
For the complex lines where you need the constant I suggest defining each constant right before its console.log

Reply:
okay, now simplified 👍